File: 2107-find-unique-binary-string/find-unique-binary-string.cpp
class Solution {
public:
    string findDifferentBinaryString(vector<string>& nums) {
        string res;
        for (int i = 0; i < nums[0].size(); i++) {
            res += (nums[i][i] == '0') ? '1' : '0';
        }
        return res;
    }
};

File: 2107-find-unique-binary-string/find-unique-binary-string.py
class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        res = []
        for i in range(len(nums)):
            res.append('1' if nums[i][i] == '0' else '0')
        return "".join(res)
        '''
        numsSet = set(nums)
        def backtrack(i, cur):
            if i == len(nums):
                res = "".join(cur)
                return res if res not in numsSet else None
            
            cur[i] = '1'
            res = backtrack(i + 1, cur)
            if res: return res
            cur[i] = '0'
        
            res = backtrack(i + 1, cur)
            if res: return res

        return backtrack(0, ['0' for n in nums])
        '''